website/client/src/components/tasks/task.vue

```diff
@@ -1071,8 +1071,10 @@ export default {
       this.scoreChecklistItem({ taskId: this.task._id, itemId: item.id });
     },
     calculateTimeTillDue () {
-      const endOfToday = moment().subtract(this.user.preferences.dayStart, 'hours').endOf('day');
-      const endOfDueDate = moment(this.task.date).endOf('day');
+      const endOfDay = time => time.subtract(this.user.preferences.dayStart, 'hours').endOf('day');
+
+      const endOfToday = endOfDay(moment());
+      const endOfDueDate = endOfDay(moment(this.task.date));
```

Comment:
I think that subtracting the same number of hours from both values and then comparing them results in the same result as not using the dayStart preference at all and might be causing the test to fail

Reply:
Weird, they succeeded locally -> Perhaps I didn't make it as server-time-independent as I had hoped.

Do note that it's not just "subtracting the same number from both sides of the equation" as what happens is that the dayStart gets subtracted -> And then the time is set to the end of the current day, meaning that the critical issue is whether the subtraction carries that specific time back to before midnight, which should be different for both times.

I'll look at this tonight.

Reply:
So now they succeeded but I really think I didn't fundamentally change anything.  I only made the test setup more explicit.  I think that's better anyway but I'm worried the test might still be flaky.  I suspect that it might be that `moment().toDate()` perhaps uses server time zone instead of the moment timezone, but that would actually be kinda weird.

Yeah, really don't know why it failed, hoping it's not intermittent.

Reply:
>  I suspect that it might be that moment().toDate() perhaps uses server time zone instead of the moment timezone, but that would actually be kinda weird.

Moment uses the timezone of the server (if it's run on a server - for example in Github actions) or the browser timezone when run in a browser. The only way to force a specific timezone is to use utcOffset https://momentjs.com/docs/#/manipulating/utc-offset/

Reply:
I don't think that the daystart should be subtracted from both dates, and for the tests we should make sure they're using a reproducible time by correctly setting the timezone

Reply:
@paglias Ah, I was thinking it would take the timezone from the ISO date, it might be that could be part of the cause.  Can adjust the code to use utcOffset.

There's probably something wrong with my mental model of how this whole thing works.  Apologies for wasting your time like this.  Having a bit of hard time juggling work, family, exercise and this, will make time to look in to this properly somewhere this week.

Reply:
No worries @benkelaar , we really appreciate your PRs , whenever you have time is going to be okay :)